### PR TITLE
docs(modal based components): refactor stories to csf and mdx formats - FE-3050

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour
This PR refactors all modal based component stories to new CSF and MDX formats:
- Alert,
- Confirm,
- Dialog,
- DialogFullScreen,
- Sidebar,
- Message, 
- Toast

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: cypress/features/regression/dialogFullScreen.feature [hunk 1]
```diff
@@ -34,7 +33,7 @@ Feature: Dialog Full Screen component
 
   @positive
   Scenario Outline: Change Dialog Full Screen children to <children>
-    Given I open default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "<nameOfObject>" object name
+    Given I open test_default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "<nameOfObject>" object name
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialogFullScreen.feature:36]:
```suggestion
    When I open test_default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "<nameOfObject>" object name
```

File: cypress/features/regression/dialogFullScreen.feature [hunk 2]
```diff
@@ -43,62 +42,50 @@ Feature: Dialog Full Screen component
 
   @positive
   Scenario: Enable background UI
-    Given I open default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "enableBackgroundUI" object name
+    Given I open test_default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "enableBackgroundUI" object name
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialogFullScreen.feature:45]:
```suggestion
    When I open test_default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "enableBackgroundUI" object name
```

File: cypress/features/regression/dialogFullScreen.feature [hunk 3]
```diff
@@ -43,62 +42,50 @@ Feature: Dialog Full Screen component
...
     Then Background UI is enabled
 
   @negative
   Scenario: Disable background UI
-    Given I open default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "enableBackgroundUIFalse" object name
+    Given I open test_default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "enableBackgroundUIFalse" object name
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialogFullScreen.feature:50]:
```suggestion
    When I open test_default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "enableBackgroundUIFalse" object name
```

File: cypress/features/regression/dialogFullScreen.feature [hunk 4]
```diff
@@ -43,62 +42,50 @@ Feature: Dialog Full Screen component
...
       And I check disableEscKey checkbox
       And I uncheck disableEscKey checkbox
-      And I open component preview
     When I hit ESC key
     Then Dialog Full Screen is not visible
 
   @positive
   Scenario: Verify that stickyFormFooter is visible
-    Given I open with_sticky_footer "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "open" object name
+    Given I open test_default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "stickyFooter" object name
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialogFullScreen.feature:70]:
```suggestion
    When I open test_default "Dialog Full Screen" component in noIFrame with "dialogFullScreen" json from "commonComponents" using "stickyFooter" object name
```

File: cypress/features/regression/message.feature [hunk 5]
```diff
@@ -3,13 +3,13 @@ Feature: Message component
 
   @positive
   Scenario: CloseIcon has correct border colour
-    Given I open default "Message" component in noIFrame with "message" json from "commonComponents" using "default" object name
-    When I click closeIcon
+    When I open test_default "Message" component in noIFrame with "message" json from "commonComponents" using "default" object name
```
Comment (KatarzynaQA) [inline cypress/features/regression/message.feature:6]:
```suggestion
    Given I open test_default "Message" component in noIFrame with "message" json from "commonComponents" using "default" object name
```

**Review by KatarzynaQA — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Comment by DipperTheDan** [comment]
Seems to be an issue with the setAttribute being null in the Alert story.

Screenshot 2020-09-14 at 14 02 29

**Comment by DipperTheDan** [comment]
Same with the Confirm story too.

Screenshot 2020-09-14 at 14 09 29

**Comment by DipperTheDan** [comment]
This is how to recreate the bug.

Sep-14-2020 14-14-35

**Comment by jamime** [comment]
@DipperTheDan issue should be fixed. The portal was attaching itself to `root` which is where the canvas storybook renders. So when the dialog was being unmounted react was destroying the storybook `<div id="root">`. Storybook was then trying to set an attribute on this element which no longer exists.

I changed 

```diff
- portalId="root"
+ portalId="portal-root"
```
I also rebased off master.

File: src/components/dialog-full-screen/dialog-full-screen.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+/>
+
+# DialogFullScreen
+A full-width and full-height dialog overlaid on top of any page.
+
+Useful to perform an action in context without navigating the user to a separate page.
+
+Whilst a standard Dialog component’s width might constrain what you can do, the full screen dialog uses the full width and height available.
+
+A good example could be importing a large volume of data, and checking for errors, in the context of an underlying Table of data.
```
Comment (sjohnson-sage) [inline src/components/dialog-full-screen/dialog-full-screen.stories.mdx:21]: Very minor but Table doesn't need a capital T here

File: src/components/message/message.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+<Meta
+  title="Message"
+  parameters={{ info: { disable: true }}}
+/>
+
+# Message
+A longer message which stays on screen to read and interpret.
+
+Presents a static message which stays on screen.
```
Comment (sjohnson-sage) [inline src/components/message/message.stories.mdx:15]: We probably don't need both this sentence and the one above, either can be removed I think

File: src/components/message/message.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+## Related Components
+
+- Simple positive or negative confirmation? [Try Flash](/?path=/story/flash--default "Flash").
+- Longer, time sensitive message that must be dismissed? [Try Toast](/?path=/docs/design-system-toast--default-story "Toast").
+- Error or warning message that interrupts activity? [Try Alert](/?path=/docs/alert--default-story "Alert").
+
+Import the component and use as shown below:
+
+```jsx
+import Message from "carbon-react/lib/components/message"
```
Comment (sjohnson-sage) [inline src/components/message/message.stories.mdx:35]: Might as well put a semi-colon on the end of these example imports so the user can just copy and paste and not have to add one in themselves

File: src/components/sidebar/sidebar.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+Choose whether a dark tint is applied behind the dialog which helps to focus the user on the sidebar, but don’t select this option if the user needs to refer back to the underlying page.
+
+## Related Components
+- Simple task in context? [Try Dialog](/?path=/docs/dialog--default-story "Try Dialog").
+- Complex task that needs more space?  [Try Dialog Full Screen](/?path=/docs/dialog-full-screen--default-story "Try Dialog Full Screen").
+
+## Quick Start
+
+Side bar is positioned on the right hand screen of the window by default.
```
Comment (sjohnson-sage) [inline src/components/sidebar/sidebar.stories.mdx:27]: "Side bar" should be "Sidebar"

File: src/components/sidebar/sidebar.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+A Sidebar header is also provided as a sub component and can be used as the first child
+of the sidebar component
+
+To render content within the SidebarHeader pass a children prop
+
+Import the components and use as shown below:
+
+```jsx
+import { Sidebar, SidebarHeader } from "carbon-react/lib/components/sidebar"
```
Comment (sjohnson-sage) [inline src/components/sidebar/sidebar.stories.mdx:41]: Same as above, a semi-colon at the end would be good here

**Review by sjohnson-sage — COMMENTED** [review]
All looks good, just a few minor comments from me

File: src/components/alert/alert.stories.mdx [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+- Longer, time sensitive message that must be dismissed? [Try Toast](/?path=/docs/design-system-toast--default-story "Try Toast").
+- Confirm or cancel an action I’ve initiated? [Try Confirm](/?path=/docs/confirm--default-story "Try Confirm").
+- Simple task in context? [Try Dialog](/?path=/docs/dialog--default-story "Try Dialog").
+
+## Quick Start
+
+Import the component and use as shown below:
+
+```jsx
+import Alert from "carbon-react/lib/components/alert"
```
Comment (DipperTheDan) [inline src/components/alert/alert.stories.mdx]: Same as Sam mentioned before. It's probably best to put a semi-colon at the end of the line so the consumer can just copy and paste the import.

File: src/components/confirm/confirm.stories.mdx [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+- Longer, time sensitive message that must be dismissed? [Try Toast](/?path=/docs/design-system-toast--default-story "Toast").
+- Error or warning message that interrupts activity? [Try Alert](/?path=/docs/alert--default-story "Alert").
+- Simple task in context? [Try Dialog](/?path=/docs/dialog--default-story "Dialog").
+
+## Quick Start
+
+Import the component and use as shown below:
+
+```jsx
+import Confirm from "carbon-react/lib/components/confirm"
```
Comment (DipperTheDan) [inline src/components/confirm/confirm.stories.mdx:37]: Semi-colon required.

**Review by KatarzynaQA — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by KatarzynaQA — APPROVED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by KatarzynaQA — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

File: src/components/message/message.stories.mdx [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+- **Success** - indicates that an activity was successful. A good example could also present the user with onward options, such as ‘View a list of items’ or ‘Create another’.
+- **Warning** - warns the user about a potential negative outcome that hasn’t happened yet.
+- The Transparent configuration is useful if you’d like the message to be more visually subtle, perhaps in a Dialog.
+
+## Related Components
+
+- Simple positive or negative confirmation? [Try Flash](/?path=/story/flash--default "Flash").
+- Longer, time sensitive message that must be dismissed? [Try Toast](/?path=/docs/design-system-toast--default-story "Toast").
+- Error or warning message that interrupts activity? [Try Alert](/?path=/docs/alert--default-story "Alert").
+
```
Comment (ianforster-sage) [inline src/components/message/message.stories.mdx:29]: Missing Quick Start heading
```suggestion
## Quick Start
```

**Review by ianforster-sage — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**